fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/LuceneAnalyzerCombinationProvider.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * LuceneAnalyzerCombinationProvider.java
+ *
+ * This source file is part of the FoundationDB open source project
+ *
+ * Copyright 2015-2022 Apple Inc. and the FoundationDB project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.apple.foundationdb.record.lucene;
+
+import org.apache.lucene.analysis.miscellaneous.PerFieldAnalyzerWrapper;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.SortedMap;
+import java.util.TreeMap;
+import java.util.stream.Collectors;
+
+/**
+ * Provide a combination of analyzers for multiple fields of one Lucene index.
+ * The combinations of analyzers for index time and query time are potentially different.
+ * The default analyzer chooser is used for all fields of one Lucene index except the fields which has overrides in the analyzer chooser per field mapping.
+ */
+public class LuceneAnalyzerCombinationProvider {
```

Comment:
The ```AnalyzerChooser``` is designed for the provision of an Lucene analyzer given a text, so it is per Lucene analyzer. The only reason we need it for now, is for the capability needed to use different one for different languages. Whereas the ```LuceneAnalyzerCombinationProvider``` is more like a single object to provide a pair of analyzers for index time and query time based on the options config, because we want to make is simple for clients to configure for the analyzers to use, without worrying about which to use on index time and which to use on query time.

Reply:
I think I like the abstraction, for the most part, and that it makes sense to encapsulate index-and-query-time-analyzers into an object other than a pair. However, once you have that abstraction, I'm not sure it makes sense to combine that with the behavior that constructs the per-field Lucene analyzers.

I also think the abstraction is fine for now, though, and something we could iterate on later if we really want to